File: api/game/Game.py
import random

from game import Settings
from game import Stats
from game import Log

class Game:
	def __init__(self):
		self.settings = Settings.Settings()
		self.stats = Stats.Stats()
		self.log = Log.Log()

		self.current_example = []
		self.setExample()

	def setExample(self):
		dif = self.settings.difficulty
		self.current_example = [self.generateExample(dif), self.generateExample(dif), self.generateOperators(self.settings.get_options())]

	def generateExample(self, difficulty):
		return random.randint(1, int(difficulty) * 5)

	def generateOperators(self, options):
		operators = []
		for option in options:
			if options[option]:
				operators.append(option)
		return random.choice(operators)

	def checkanswer(self, answer, time):
		OPERATORS = {
            "plus": "+",
            "minus": "-",
            "multiply": "*",
            "divide": "/"
        }

		result = eval(str(self.current_example[0]) + str(OPERATORS[self.current_example[2]]) + str(self.current_example[1]))

		self.stats.addGame()
		self.log.addLog(self.current_example, answer, result, time, self.settings.difficulty)
		if answer == float(result):
			self.stats.addWin()
			return True
		else:
			self.stats.addLoss()
			return result

File: src/game/Game.js
import {Settings} from "./Settings.js";
import {Stats} from "./Stats.js";
import {Log} from "./Log.js";


    class Game {
    constructor() {
        this.setting = new Settings();
        this.stats = new Stats();
        this.log = new Log();

        this.current_example = this.setExample()
    }

    setExample() {
        let dif = this.setting.getDifficulty()
        return [this.generateExample(dif), this.generateExample(dif), this.generateOperators(this.setting.getOptions())]
    }


    generateExample(difficulty) {
        return Math.floor(Math.random() * (difficulty * 5));
    }

    generateOperators(options) {
        // {"plus": true, "minus": false, "multiply": false, "divide": false}
        // Добавляем в массив знаки  которые включены и рандомим знак
        let operators = [];
        for (let key in options) {
            if (options[key]) {
                operators.push(key);
            }
        }
        return operators[Math.floor(Math.random() * operators.length)];
    }

    checkAnswer(answer) {
        const OPERATORS = {
            "plus": "+",
            "minus": "-",
            "multiply": "*",
            "divide": "/"
        }
        this.stats.addGame()
        let result = eval(this.current_example[0] + OPERATORS[this.current_example[2]] + this.current_example[1]);
        if (answer === result) {
            this.stats.addWin();

            this.log.addLog(this.current_example, true, result, result, this.setting.getDifficulty());
            return true;
        }
        this.stats.addLoss();
        this.log.addLog(this.current_example, false, result, result, this.setting.getDifficulty());
        return false;

    }
}


export default Game;
